encode.go

```diff
@@ -132,6 +132,20 @@ func (e *MarshalerError) Unwrap() error {
 	return e.err
 }
 
+type TranscodeError struct {
+	err            error
+	rtype          reflect.Type
+	source, target string
```

Comment:
Maybe rename these fields to be more explicit.

```suggestion
	sourceFormat, targetFormat string
```